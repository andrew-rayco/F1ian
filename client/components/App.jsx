import React from 'react'

import RunRace from './RunRace'
const App = raceData => {
  return (
    <div>
      <h1>Visualise</h1>
      <RunRace raceData={raceData.raceData} />
    </div>
  )
}



export default App
