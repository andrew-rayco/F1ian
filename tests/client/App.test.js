import React from 'react'
import { shallow, mount } from 'enzyme'
import App from '../../client/components/App'

describe('App', () => {
  let props = { location: { pathname: '1/2/3/4/5'}, year: 1985 }
  const app = shallow(<App {...props} />)

  it('renders correctly', () => {
    expect(app).toMatchSnapshot()
  })

  it('contains a container div', () => {
    expect(app.find('.container').exists()).toBe(true)
  })
})
