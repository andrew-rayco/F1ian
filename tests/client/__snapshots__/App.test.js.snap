// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App renders correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <App
    location={
      Object {
        "pathname": "1/2/3/4/5",
      }
    }
    year={1985}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": <div
        className="container"
      >
        <Route
          component={[Function]}
          path="/"
        />
        <Route
          component={[Function]}
          exact={true}
          path="/"
        />
        <Route
          component={[Function]}
          exact={true}
          path="/circuits"
        />
        <Route
          component={[Function]}
          exact={true}
          path="/season/:id"
        />
        <Route
          component={[Function]}
          path="/season/:id/:raceId/qualifying"
        />
        <Route
          component={[Function]}
          path="/season/:id/:raceId/grid"
        />
        <Route
          component={[Function]}
          path="/season/:id/:raceId/visualise"
        />
        <Route
          component={[Function]}
          path="/season/:id/:raceId/results"
        />
      </div>,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <Route
            component={[Function]}
            path="/"
          />,
          <Route
            component={[Function]}
            exact={true}
            path="/"
          />,
          <Route
            component={[Function]}
            exact={true}
            path="/circuits"
          />,
          <Route
            component={[Function]}
            exact={true}
            path="/season/:id"
          />,
          <Route
            component={[Function]}
            path="/season/:id/:raceId/qualifying"
          />,
          <Route
            component={[Function]}
            path="/season/:id/:raceId/grid"
          />,
          <Route
            component={[Function]}
            path="/season/:id/:raceId/visualise"
          />,
          <Route
            component={[Function]}
            path="/season/:id/:raceId/results"
          />,
        ],
        "className": "container",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "component": [Function],
            "path": "/",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "component": [Function],
            "exact": true,
            "path": "/",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "component": [Function],
            "exact": true,
            "path": "/circuits",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "component": [Function],
            "exact": true,
            "path": "/season/:id",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "component": [Function],
            "path": "/season/:id/:raceId/qualifying",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "component": [Function],
            "path": "/season/:id/:raceId/grid",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "component": [Function],
            "path": "/season/:id/:raceId/visualise",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "component": [Function],
            "path": "/season/:id/:raceId/results",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "div",
    },
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": <div
          className="container"
        >
          <Route
            component={[Function]}
            path="/"
          />
          <Route
            component={[Function]}
            exact={true}
            path="/"
          />
          <Route
            component={[Function]}
            exact={true}
            path="/circuits"
          />
          <Route
            component={[Function]}
            exact={true}
            path="/season/:id"
          />
          <Route
            component={[Function]}
            path="/season/:id/:raceId/qualifying"
          />
          <Route
            component={[Function]}
            path="/season/:id/:raceId/grid"
          />
          <Route
            component={[Function]}
            path="/season/:id/:raceId/visualise"
          />
          <Route
            component={[Function]}
            path="/season/:id/:raceId/results"
          />
        </div>,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <Route
              component={[Function]}
              path="/"
            />,
            <Route
              component={[Function]}
              exact={true}
              path="/"
            />,
            <Route
              component={[Function]}
              exact={true}
              path="/circuits"
            />,
            <Route
              component={[Function]}
              exact={true}
              path="/season/:id"
            />,
            <Route
              component={[Function]}
              path="/season/:id/:raceId/qualifying"
            />,
            <Route
              component={[Function]}
              path="/season/:id/:raceId/grid"
            />,
            <Route
              component={[Function]}
              path="/season/:id/:raceId/visualise"
            />,
            <Route
              component={[Function]}
              path="/season/:id/:raceId/results"
            />,
          ],
          "className": "container",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "component": [Function],
              "path": "/",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "component": [Function],
              "exact": true,
              "path": "/",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "component": [Function],
              "exact": true,
              "path": "/circuits",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "component": [Function],
              "exact": true,
              "path": "/season/:id",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "component": [Function],
              "path": "/season/:id/:raceId/qualifying",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "component": [Function],
              "path": "/season/:id/:raceId/grid",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "component": [Function],
              "path": "/season/:id/:raceId/visualise",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "component": [Function],
              "path": "/season/:id/:raceId/results",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
        ],
        "type": "div",
      },
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactFifteenAdapter {
      "options": Object {
        "supportPrevContextArgumentOfComponentDidUpdate": true,
      },
    },
  },
}
`;
